p2p/src/net/mock/types.rs

```diff
@@ -68,7 +94,28 @@ pub enum PubSubEvent {
     },
 }
 
-pub enum SyncingEvent {}
+#[derive(Copy, Clone, PartialEq, Eq, Hash, Debug, Encode, Decode, Default)]
+pub struct MockRequestId(u64);
+
+impl MockRequestId {
+    pub fn new(request_id: u64) -> Self {
+        Self(request_id)
+    }
+}
+
+impl std::ops::Deref for MockRequestId {
+    type Target = u64;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
```

Comment:
This will probably expose some operations that don't make sense on IDs. How many u64 operations are actually nneded, besides comparisons which are auto-derived anyway? Possibly just exposing required operations one by one would not be that much more code.

Reply:
Done

Also simplified the request ID allocation to just use one common counter for all outbound requests and increment it once for each request. No need to deal with peer-specific zones, it was part of previous design.